tests/Generator.hs

```diff
@@ -106,4 +106,10 @@ offerMessages :: MonadGen m => m F.Offer
 offerMessages = Gen.choice
   [ F.Message <$> (Gen.text (Range.linear 0 1024) Gen.unicode)
   , F.File <$> toS <$> (Gen.text (Range.linear 0 100) Gen.alphaNum)  <*> (fromIntegral <$> (Gen.int (Range.linear 0 maxBound)))
+  , F.Directory
+    <$> (toS <$> (Gen.text (Range.linear 0 100) Gen.alphaNum))
+    <*> (toS <$> (Gen.text (Range.linear 0 100) Gen.alphaNum))
+    <*> (fromIntegral <$> (Gen.int (Range.linear 0 maxBound)))
+    <*> (fromIntegral <$> (Gen.int (Range.linear 0 maxBound)))
+    <*> (fromIntegral <$> (Gen.int (Range.linear 0 maxBound)))
```

Comment:
These three fields are `Int64` and can contain negative integers.  I think `(Gen.int (Range.linear 0 maxBound))` will only generate `[0, maxBound]`.  Maybe the types of these fields should instead be `Data.Word.Word64`?  A negative value for any of them seems nonsensical.  On the other hand, I don't like the overflow behavior of `Word64`.  On the first hand again, `Int64` has the same behavior!  So `Word64` might still be slightly better.  But I wonder what the Aeson behavior of decoding "-1"` into a Word64 is.  An error?  A overflow?

So unfortunately I'm not really sure what I'm suggesting here.  Apart from an idea that the generator for the type should exercise the full domain of the type (particularly the crazy parts).

Reply:
Natural numbers are apparently a thing.

http://hackage.haskell.org/package/base-4.11.1.0/docs/Numeric-Natural.html

and also

http://hackage.haskell.org/package/naturals-0.2.0.2/docs/Data-Natural.html

Reply:
> But I wonder what the Aeson behavior of decoding "-1"` into a Word64 is. An error? A overflow?

I am not sure either. It definitely seem wrong to use a negative number for these fields as they all represent sizes. `Numeric.Natural` is nice.